tensorflow/lite/experimental/shlo/ops/dot_general_test.cc

```diff
@@ -0,0 +1,1894 @@
+      .rhs_contracting_dimensions = rhs_contracting_dimensions,
+      .precision_configs = precision_configs});
+
+  Vector<int64_t> expected_data_int{13};
+  Vector<StorageT> expected_data(expected_data_int.begin(),
+                                 expected_data_int.end());
+
+  ASSERT_OK(Prepare(op, lhs, rhs, output_tensor));
+  ASSERT_OK(Evaluate(op, lhs, rhs, output_tensor));
+  EXPECT_THAT(output_data, Pointwise(Eq(), expected_data));
+}
+
+using kF32TestTypes = ::testing::Types<TestParam<DataType::kF32>>;
+template <class T>
+struct NonQuantizedkF32DotGeneralTest : ::testing::Test {};
+
+TYPED_TEST_SUITE(NonQuantizedkF32DotGeneralTest, kF32TestTypes, TestParamNames);
+
+TYPED_TEST(NonQuantizedkF32DotGeneralTest, kF32TestTypesTensorsWork1) {
+  using StorageT = typename TypeParam::StorageT;
+
+  const Shape shape_lhs({4});
+  const Shape shape_rhs({4});
+  const Shape shape_r({1});
+
+  Vector<StorageT> lhs_data{-1.73818827, 6.32115507, 2.81545162, -1.37914991};
+  Vector<StorageT> rhs_data{-4.02553225, -2.70646834, 3.14252234, 1.59961236};
+  Vector<StorageT> output_data(shape_r.NumElements());
+  Vector<int64_t> lhsb_dim{};
+  Vector<int64_t> rhsb_dim{};
+  Vector<int64_t> lhsc_dim{0};
+  Vector<int64_t> rhsc_dim{0};
+  absl::Span<int64_t> lhs_batching_dimensions(lhsb_dim);
+  absl::Span<int64_t> rhs_batching_dimensions(rhsb_dim);
+  absl::Span<int64_t> lhs_contracting_dimensions(lhsc_dim);
+  absl::Span<int64_t> rhs_contracting_dimensions(rhsc_dim);
+
+  Tensor lhs{.type = TensorType{.shape = shape_lhs,
+                                .element_type = TypeParam::kStorage},
+             .data = lhs_data.data()};
+  Tensor rhs{.type = TensorType{.shape = shape_rhs,
+                                .element_type = TypeParam::kStorage},
+             .data = rhs_data.data()};
+  Tensor output_tensor{
+      .type = TensorType{.shape = shape_r, .element_type = TypeParam::kStorage},
+      .data = output_data.data()};
+  std::array<PrecisionTypes, 2> precision_configs = {PrecisionTypes::DEFAULT,
+                                                     PrecisionTypes::DEFAULT};
+
+  auto op = Create(DotGeneralOp::Attributes{
+      .lhs_batching_dimensions = lhs_batching_dimensions,
+      .rhs_batching_dimensions = rhs_batching_dimensions,
+      .lhs_contracting_dimensions = lhs_contracting_dimensions,
+      .rhs_contracting_dimensions = rhs_contracting_dimensions,
+      .precision_configs = precision_configs});
+
+  Vector<StorageT> expected_data = {-3.46935892};
+
+  ASSERT_OK(Prepare(op, lhs, rhs, output_tensor));
+  ASSERT_OK(Evaluate(op, lhs, rhs, output_tensor));
+  EXPECT_THAT(output_data, Pointwise(FloatNear(0.00001), expected_data));
+}
+
+TYPED_TEST(NonQuantizedkF32DotGeneralTest, kF32TestTypesTensorsWork2) {
+  using StorageT = typename TypeParam::StorageT;
+
+  const Shape shape_lhs({8, 4, 3, 3, 4});
+  const Shape shape_rhs({4, 8, 3, 4, 2});
+  const Shape shape_r({8, 4, 3, 2});
+
+  Vector<StorageT> lhs_data{
```

Comment:
I'd generally suggest moving the test data to a separate file with inline constexpr arrays that can be referenced from the tests. The large initialization blocks become distracting from reading the test bodies themselves.

Reply:
We have made the required changes, will update the same in the next commit.